### PR TITLE
add restrictions to requests when xjoin is the data source

### PR DESCRIPTION
**Pull request by ryandillinfelton**

### CODE REVIEW

File: tests/test_xjoin.py [hunk 1]
```diff
@@ -1460,3 +1460,72 @@ def test_query_system_profile_sap_sids_with_search(
     graphql_system_profile_sap_sids_query_with_response.assert_called_once_with(
         SAP_SIDS_QUERY, {"hostFilter": {"OR": mocker.ANY}, "filter": {"search": {"regex": ".*C2.*"}}}
     )
+
+def test_query_hosts_system_identity(
+    mocker, subtests, query_source_xjoin, graphql_query_empty_response, api_get
+):
+    url = build_hosts_url()
+
+    response_status, response_data = api_get(url, identity_type="System")
```
Comment (thearifismail) [inline tests/test_xjoin.py:1469]: unused variable `response_data`

File: tests/test_xjoin.py [hunk 2]
```diff
@@ -1460,3 +1460,72 @@ def test_query_system_profile_sap_sids_with_search(
...
+        },
+    )
+
+
+def test_query_tags_system_identity(
+    mocker, subtests, query_source_xjoin, graphql_tag_query_empty_response, api_get
+):
+    url = build_tags_url()
+
+    response_status, response_data = api_get(url, identity_type="System")
```
Comment (thearifismail) [inline tests/test_xjoin.py:1490]: unused variable `response_data`

File: tests/test_xjoin.py [hunk 3]
```diff
@@ -1460,3 +1460,72 @@ def test_query_system_profile_sap_sids_with_search(
...
+        },
+    )
+
+
+def test_query_system_profile_sap_sids_system_identity(
+    mocker, subtests, query_source_xjoin, graphql_system_profile_sap_sids_query_with_response, api_get
+):
+    url = build_system_profile_sap_sids_url()
+
+    response_status, response_data = api_get(url, identity_type="System")
```
Comment (thearifismail) [inline tests/test_xjoin.py:1511]: unused variable `response_data`

**Comment by kruai** [comment]
@ryandillinfelton it sounded like this is ready for review, so I'm going to take it out of draft status

**Review by thearifismail — APPROVED** [review]
there are numerous warnings for unused `response_data`.  Replacing the unused `response_data` with underscore takes care of the warnings and a more proper way.  It is more elegant to remove the unused variables, but if you want to leave them because of time constraint, I am okay with it.

Therefore I am going to `Approve` the PR.

**Comment by ryandillinfelton** [comment]
> there are numerous warnings for unused `response_data`. Replacing the unused `response_data` with underscore takes care of the warnings and a more proper way. It is more elegant to remove the unused variables, but if you want to leave them because of time constraint, I am okay with it.
> 
> Therefore I am going to `Approve` the PR.

I think we should just go ahead and merge it now if the response_data warning is the only issue. I think it would be a good idea to have a sperate PR cleaning up all the tests that have the warning since iirc it's very prevalent.

**Review by thearifismail — APPROVED** [review]

**Review by kruai — APPROVED** [review]

**Comment by kruai** [comment]
Looks like there are some conflicts that need resolving @ryandillinfelton

**Comment by ryandillinfelton** [comment]
@kruai @thearifismail conflicts resolved :trumpet:

**Comment by kruai** [comment]
Great! I'm merging it.

**Pull request merged**